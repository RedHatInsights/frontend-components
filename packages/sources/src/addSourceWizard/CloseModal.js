import React from 'react';
import PropTypes from 'prop-types';
import { Modal, Button, TextContent, Text, TextVariants } from '@patternfly/react-core';
import { ExclamationTriangleIcon } from '@patternfly/react-icons';

const CloseModal = ({ onExit, onStay, isOpen, title, exitTitle, stayTitle, description }) => (
    <Modal
        variant="small"
        title={title}
        aria-label="Close add source wizard"
        header={
            <TextContent>
                <ExclamationTriangleIcon size="lg" className="ins-c-source__warning-icon" />
                <Text component={TextVariants.h1}>
                    {title}
                </Text>
            </TextContent>
        }
        isOpen={isOpen}
        onClose={onStay}
        actions={[
            <Button key="confirm" variant="primary" id="on-exit-button" onClick={onExit}>
                {exitTitle}
            </Button>,
            <Button key="cancel" variant="link" id="on-stay-button" onClick={onStay}>
                {stayTitle}
            </Button>
        ]}
    >
        {description}
    </Modal>
);

CloseModal.propTypes = {
    onExit: PropTypes.func.isRequired,
    onStay: PropTypes.func.isRequired,
    isOpen: PropTypes.bool.isRequired,
    title: PropTypes.node,
    exitTitle: PropTypes.node,
    stayTitle: PropTypes.node,
    description: PropTypes.node
};

CloseModal.defaultProps = {
    title: 'Exit source creation?',
    exitTitle: 'Exit',
    stayTitle: 'Stay',
    description: 'All inputs will be discarded.'
};

export default CloseModal;
