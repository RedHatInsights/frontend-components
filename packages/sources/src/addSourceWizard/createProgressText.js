const createProgressText = (formData) => {
    let progressTexts = [];
    let step = 1;

    const authAndApp = formData.endpoint && formData.application && formData.application.application_type_id;

    progressTexts.push('Step 1: sending source data');
    if (authAndApp) {
        progressTexts.push(`Step ${++step}: sending endpoint and application data`);
    } else if ((formData.endpoint) && !(formData.application && formData.application.application_type_id)) {
        progressTexts.push(`Step ${++step}: sending endpoint data`);
    } else if (!(formData.endpoint) && (formData.application && formData.application.application_type_id)) {
        progressTexts.push(`Step ${++step}: sending application data`);
    }

    if (formData.endpoint) {
        progressTexts.push(`Step ${++step}: sending authentication data`);
    }

    if (formData.credentials || formData.billing_source) {
        progressTexts.push(`Step ${++step}: sending cost management data and connecting authentication and application`);
    } else if (authAndApp) {
        progressTexts.push(`Step ${++step}: connecting authentication and application`);
    }

    progressTexts.push('Completed');

    return progressTexts;
};

export default createProgressText;
