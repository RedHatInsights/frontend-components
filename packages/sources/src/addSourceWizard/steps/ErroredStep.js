import React from 'react';
import PropTypes from 'prop-types';

import { EmptyStateBody, EmptyState, EmptyStateIcon, Title, Button, EmptyStateSecondaryActions, EmptyStateVariant } from '@patternfly/react-core';
import { ErrorCircleOIcon } from '@patternfly/react-icons';

const ErroredStep = ({ onClose, onRetry }) => <div className="pf-l-bullseye">
    <EmptyState variant={ EmptyStateVariant.full }>
        <EmptyStateIcon icon={ ErrorCircleOIcon } color="var(--pf-global--danger-color--100)" />
        <Title headingLevel="h5" size="lg">
      Configuration unsuccessful
        </Title>
        <EmptyStateBody>
      Something went wrong. Your source has not been successfully added.
        </EmptyStateBody>
        <Button variant="primary" onClick={ onClose }>Back to my application</Button>
        <EmptyStateSecondaryActions>
            <Button variant="link" onClick={ onRetry }>Retry</Button>
        </EmptyStateSecondaryActions>
    </EmptyState>
</div>;

ErroredStep.propTypes = {
    onClose: PropTypes.func.isRequired,
    onRetry: PropTypes.func.isRequired
};

export default ErroredStep;
