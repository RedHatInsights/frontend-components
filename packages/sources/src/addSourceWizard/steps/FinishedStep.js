import React from 'react';
import PropTypes from 'prop-types';

import { EmptyStateBody, EmptyState, EmptyStateIcon, Title, Button, EmptyStateSecondaryActions, EmptyStateVariant } from '@patternfly/react-core';
import { CheckCircleIcon } from '@patternfly/react-icons';

const FinishedStep = ({ onClose, successfulMessage }) => <div className="pf-l-bullseye">
    <EmptyState variant={ EmptyStateVariant.full }>
        <EmptyStateIcon icon={ CheckCircleIcon } color="var(--pf-global--success-color--100)" />
        <Title headingLevel="h5" size="lg">
      Configuration successful
        </Title>
        <EmptyStateBody>
            { successfulMessage }
        </EmptyStateBody>
        <Button variant="primary" onClick={ onClose }>Back to my application</Button>
        <EmptyStateSecondaryActions>
            <a href='/hybrid/settings/sources'>
                <Button variant="link">Take me to sources</Button>
            </a>
        </EmptyStateSecondaryActions>
    </EmptyState>
</div>;

FinishedStep.propTypes = {
    onClose: PropTypes.func.isRequired,
    successfulMessage: PropTypes.node.isRequired
};

export default FinishedStep;
