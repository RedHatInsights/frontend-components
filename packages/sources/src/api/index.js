import axios from 'axios';
import { DefaultApi as SourcesDefaultApi } from '@redhat-cloud-services/sources-client';
import { Base64 } from 'js-base64';
import { postBillingSource } from './billingSource';

const calculateApiBase = b => (
    (b.endsWith('/') && `${b}v1.0`) || `${b}/sources/v1.0`
);

export const SOURCES_API_BASE = calculateApiBase(process.env.BASE_PATH || '');

const axiosInstance = axios.create(
    process.env.FAKE_IDENTITY ? {
        headers: {
            common: {
                'x-rh-identity': Base64.encode(
                    JSON.stringify(
                        {
                            identity: { account_number: process.env.FAKE_IDENTITY }
                        }
                    )
                )
            }
        }
    } : {}
);

axiosInstance.interceptors.request.use(async (config) => {
    await window.insights.chrome.auth.getUser();
    return config;
});
axiosInstance.interceptors.response.use(response => response.data || response);
axiosInstance.interceptors.response.use(null, error => { throw { ...error.response }; });

export { axiosInstance };

let apiInstance;

export const getSourcesApi = () =>
    apiInstance || (apiInstance = new SourcesDefaultApi(undefined, SOURCES_API_BASE, axiosInstance));

export const doLoadSourceTypes = () =>
    getSourcesApi().listSourceTypes().then(data => ({ sourceTypes: data.data }));

export const doLoadApplicationTypes = () =>
    getSourcesApi().listApplicationTypes().then(data => ({ applicationTypes: data.data }));

const parseUrl = url => {
    if (!url) {
        return ({});
    }

    try {
        const u = new URL(url);
        return {
            scheme: u.protocol.replace(/:$/, ''),
            host: u.hostname,
            port: u.port,
            path: u.pathname
        };
    } catch (error) {
        console.log(error);
        return ({});
    }
};

/*
     * If there's an URL in the formData, parse it and use it,
     * else use individual fields (scheme, host, port, path).
     */

const urlOrHost = formData => formData.url ? parseUrl(formData.url) : formData;

export function doCreateSource(formData, sourceTypes) {
    const source_type_id = sourceTypes.find((x) => x.name === formData.source_type).id;

    return getSourcesApi().createSource({ ...formData.source, source_type_id }).then((sourceDataOut) => {
        const { scheme, host, port, path } = urlOrHost(formData);

        const endPointPort = parseInt(port, 10);

        const endpointData = {
            ...formData.endpoint,
            default: true,
            source_id: sourceDataOut.id,
            scheme,
            host,
            port: isNaN(endPointPort) ? undefined : endPointPort,
            path
        };

        const promises = [ getSourcesApi().createEndpoint(endpointData) ];

        if (formData.application.application_type_id) {
            const applicationData = {
                ...formData.application,
                source_id: sourceDataOut.id
            };

            promises.push(getSourcesApi().createApplication(applicationData));
        }

        return Promise.all(promises).then(([ endpointDataOut, applicationDataOut = undefined ]) => {
            const authenticationData = {
                ...formData.authentication,
                resource_id: endpointDataOut.id,
                resource_type: 'Endpoint'
            };

            return getSourcesApi().createAuthentication(authenticationData).then(() => {
                const source = {
                    ...sourceDataOut,
                    endpoint: [ endpointDataOut ],
                    applications: [ applicationDataOut ]
                };

                if (formData.billing_source) {
                    const billingSourceData = {
                        billing_source: formData.billing_source,
                        source_id: sourceDataOut.id
                    };

                    return postBillingSource(billingSourceData).then(() => source)
                    .catch(error => { throw { error };});
                }

                return source;
            }, (error) => {
                console.error('Authentication creation failure:', error);
                throw { error };
            });
        }).catch(e => {
            console.error(e);
            throw { error: e };
        });
    }, (error) => {
        console.error('Source creation failure:', error);
        throw { error };
    });
}

export const doLoadAllApplications = () => getSourcesApi().listApplicationTypes().then(data => data.data);

export const findSource = (name) => getSourcesApi().postGraphQL({
    query: `{ sources(filter: {name: "${name}"})
        { id, name }
    }`
});
