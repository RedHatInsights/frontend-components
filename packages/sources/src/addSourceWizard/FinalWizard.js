import React from 'react';
import PropTypes from 'prop-types';
import { Wizard } from '@patternfly/react-core';

import FinishedStep from './steps/FinishedStep';
import ErroredStep from './steps/ErroredStep';
import LoadingStep from './steps/LoadingStep';

const FinalWizard = ({ afterSubmit, afterError, isFinished, isErrored, onRetry, successfulMessage, hideSourcesButton, returnButtonTitle }) =>
    <Wizard
        isOpen={ true }
        onClose={ isFinished ? afterSubmit : afterError }
        title="Add a source"
        description="Connect an external source to Red Hat Cloud Services"
        steps={ [{
            name: 'Finish',
            component: isFinished ?
                <FinishedStep
                    onClose={ afterSubmit }
                    successfulMessage={ successfulMessage }
                    hideSourcesButton={ hideSourcesButton }
                    returnButtonTitle={ returnButtonTitle }
                /> :
                isErrored ?
                    <ErroredStep onRetry={ onRetry } onClose={ afterError } returnButtonTitle={ returnButtonTitle }/>
                    : <LoadingStep customText='Source is being created'/>,
            isFinishedStep: true
        }] } />;

FinalWizard.propTypes = {
    afterSubmit: PropTypes.func.isRequired,
    afterError: PropTypes.func.isRequired,
    onRetry: PropTypes.func.isRequired,
    isFinished: PropTypes.bool.isRequired,
    isErrored: PropTypes.bool.isRequired,
    successfulMessage: PropTypes.node.isRequired,
    hideSourcesButton: PropTypes.bool.isRequired,
    returnButtonTitle: PropTypes.node.isRequired
};

export default FinalWizard;
