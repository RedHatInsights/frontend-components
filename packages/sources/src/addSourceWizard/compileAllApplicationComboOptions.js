import React from 'react';

import { getActiveVendor, NO_APPLICATION_VALUE, REDHAT_VENDOR } from '../utilities/stringConstants';

import { COST_MANAGEMENT_APP_NAME, CLOUD_METER_APP_NAME } from '../api/constants';
import { Label } from '@patternfly/react-core';
import SubWatchDescription from './descriptions/SubWatchDescription';

export const descriptionMapper = (type, intl) => ({
    [COST_MANAGEMENT_APP_NAME]: intl.formatMessage({
        id: 'cost.app.description',
        defaultMessage: 'Analyze, forecast, and optimize your Red Hat OpenShift cluster costs in hybrid cloud environments.'
    }),
    [CLOUD_METER_APP_NAME]: <SubWatchDescription id={type.id} />
}[type.name]);

export const labelMapper = (type, intl) => ({
    [CLOUD_METER_APP_NAME]: <span className="ins-c-sources__wizard--rhel-mag-label">RHEL management <Label className="pf-u-ml-sm" color="purple">{
        intl.formatMessage({ id: 'sub.bundle', defaultMessage: 'Bundle' })
    }</Label></span>
}[type.name]);

export const compileAllApplicationComboOptions = (applicationTypes, intl) => (
    [
        ...applicationTypes.sort((a, b) => a.display_name.localeCompare(b.display_name)).map(t => ({
            value: t.id,
            label: labelMapper(t, intl) || t.display_name,
            description: descriptionMapper(t, intl)
        })),
        ...(getActiveVendor() !== REDHAT_VENDOR ? [{
            label: intl.formatMessage({
                id: 'wizard.noApplication',
                defaultMessage: 'No application'
            }),
            value: NO_APPLICATION_VALUE
        }] : [])
    ]
);
