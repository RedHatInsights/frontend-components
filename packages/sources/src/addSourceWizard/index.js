import React from 'react';
import { Button } from '@patternfly/react-core';
import PropTypes from 'prop-types';

import Form from './SourceAddModal';
import FinalWizard from './FinalWizard';

import { doCreateSource } from '../api/index';

const initialValues = (initialValues) => ({
    isSubmitted: false,
    isFinished: false,
    isErrored: false,
    values: initialValues,
    createdSource: {}
});

class AddSourceWizard extends React.Component {
    state = initialValues(this.props.initialValues);

    setOnSubmitState = (values) => this.setState({
        isSubmitted: true,
        values
    })

    onSubmit = (formValues, sourceTypes) => {
        this.setOnSubmitState(formValues);
        return doCreateSource(formValues, sourceTypes).then((data) => {
            this.setState({ isFinished: true, createdSource: data });
        })
        .catch(() => {
            this.setState({ isErrored: true });
        });
    }

    afterSubmit = () => {
        const { afterSuccess, onClose } = this.props;

        onClose();
        this.setState({ ...initialValues });
        afterSuccess(this.state.createdSource);
    }

    onRetry = () => this.setState({
        isErrored: false,
        isSubmitted: false
    })

    onCancel = () => {
        this.setState(initialValues);
        this.props.onClose();
    }

    render() {
        const {
            successfulMessage,
            isOpen,
            sourceTypes,
            applicationTypes,
            disableAppSelection,
            hideSourcesButton,
            returnButtonTitle,
            disableHardcodedSchemas
        } = this.props;
        const { isErrored, isFinished, isSubmitted, values } = this.state;

        if (!isOpen) {
            return null;
        }

        if (!isSubmitted) {
            return <Form
                values={ values }
                onSubmit={ this.onSubmit }
                onCancel={ this.onCancel }
                sourceTypes={ sourceTypes }
                applicationTypes={ applicationTypes }
                disableAppSelection={ disableAppSelection }
                disableHardcodedSchemas={ disableHardcodedSchemas }
            />;
        }

        return <FinalWizard
            afterSubmit={ this.afterSubmit }
            afterError={ this.onCancel }
            isFinished={ isFinished }
            isErrored={ isErrored }
            onRetry={ this.onRetry }
            successfulMessage={ successfulMessage }
            hideSourcesButton={ hideSourcesButton }
            returnButtonTitle={ returnButtonTitle }
        />;
    }
}

AddSourceWizard.propTypes = {
    afterSuccess: PropTypes.func,
    sourceTypes: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        product_name: PropTypes.string.isRequired,
        schema: PropTypes.shape({
            authentication: PropTypes.array,
            endpoint: PropTypes.object
        })
    })),
    applicationTypes: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        display_name: PropTypes.string.isRequired
    })),
    onClose: PropTypes.func.isRequired,
    isOpen: PropTypes.bool.isRequired,
    successfulMessage: PropTypes.node,
    initialValues: PropTypes.shape({
        [PropTypes.string]: PropTypes.oneOf([ PropTypes.string, PropTypes.array, PropTypes.number, PropTypes.bool ])
    }),
    disableAppSelection: PropTypes.bool,
    hideSourcesButton: PropTypes.bool,
    returnButtonTitle: PropTypes.node,
    disableHardcodedSchemas: PropTypes.bool
};

AddSourceWizard.defaultProps = {
    afterSuccess: () => {},
    sourceTypes: undefined,
    applicationTypes: undefined,
    successfulMessage: 'Your source has been successfully added.',
    initialValues: {},
    disableAppSelection: false,
    hideSourcesButton: false,
    returnButtonTitle: 'Go back to sources',
    disableHardcodedSchemas: false
};

class AddSourceButton extends React.Component {
    state = {
        isOpen: false
    }

    setIsOpen = (value) => this.setState({
        isOpen: value
    })

    render() {
        return <React.Fragment>
            <Button variant='primary' onClick={ () => this.setState({ isOpen: true }) }>Add a source</Button>
            <AddSourceWizard isOpen={ this.state.isOpen } onClose={ () => this.setIsOpen(false) } { ...this.props }/>
        </React.Fragment>;
    }
}

export { AddSourceButton, AddSourceWizard };
