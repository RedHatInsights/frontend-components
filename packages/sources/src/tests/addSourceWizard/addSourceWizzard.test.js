import React from 'react';
import { shallow, mount } from 'enzyme';
import { Button } from '@patternfly/react-core';

import { AddSourceWizard } from '../../addSourceWizard/index';
import Form from '../../addSourceWizard/SourceAddModal';
import FormRenderer from '../../sourceFormRenderer/index';
import Modal from '../../addSourceWizard/SourceAddModal';
import FinalWizard from '../../addSourceWizard/FinalWizard';
import FinishedStep from '../../addSourceWizard/steps/FinishedStep';
import ErroredStep from '../../addSourceWizard/steps/ErroredStep';

import sourceTypes from '../helpers/sourceTypes';
import applicationTypes from '../helpers/applicationTypes';
import * as dependency from '../../api/index';

describe('AddSourceWizard', () => {
    let initialProps;

    beforeEach(() => {
        initialProps = {
            isOpen: true,
            sourceTypes,
            applicationTypes,
            onClose: jest.fn()
        };
    });

    it('renders correctly with sourceTypes', () => {
        const wrapper = shallow(<AddSourceWizard { ...initialProps }/>);
        expect(wrapper.find(Form)).toHaveLength(1);
        expect(wrapper.find(Modal)).toHaveLength(1);
    });

    it('renders correctly without sourceTypes', () => {
        const wrapper = shallow(<AddSourceWizard { ...initialProps } sourceTypes={ undefined }/>);
        expect(wrapper.find(Form)).toHaveLength(1);
        expect(wrapper.find(Modal)).toHaveLength(1);
    });

    it('show finished step after filling the form', (done) => {
        dependency.doCreateSource = jest.fn(() => new Promise((resolve) => resolve('ok')));
        dependency.findSource = jest.fn(() => Promise.resolve({ data: { sources: [] } }));

        const wrapper = mount(<AddSourceWizard { ...initialProps }/>);
        const form = wrapper.find(FormRenderer).children().children().instance().form;

        form.change('source.name', 'nameee');
        form.change('source_type', 'openshift');

        setTimeout(() => {
            form.submit().then(() => {
                wrapper.update();
                expect(wrapper.find(FinalWizard)).toHaveLength(1);
                expect(wrapper.find(FinishedStep)).toHaveLength(1);
                expect(wrapper.find(ErroredStep)).toHaveLength(0);
                done();
            });
        }, 1000);
    });

    it('pass created source to afterSuccess function', (done) => {
        const afterSubmitMock = jest.fn();
        dependency.doCreateSource = jest.fn(() => new Promise((resolve) => resolve({ name: 'source' })));
        dependency.findSource = jest.fn(() => Promise.resolve({ data: { sources: [] } }));

        const wrapper = mount(<AddSourceWizard { ...initialProps } afterSuccess={ afterSubmitMock }/>);
        const form = wrapper.find(FormRenderer).children().children().instance().form;

        form.change('source.name', 'nameee');
        form.change('source_type', 'openshift');

        setTimeout(() => {
            form.submit().then(() => {
                wrapper.update();
                wrapper.find(Button).at(0).simulate('click');
                wrapper.update();

                expect(afterSubmitMock).toHaveBeenCalledWith({ name: 'source' });
                done();
            });
        }, 1000);
    });

    it('show error step after failing the form', (done) => {
        dependency.doCreateSource = jest.fn(() => new Promise((_resolve, reject) => reject('fail')));

        const wrapper = mount(<AddSourceWizard { ...initialProps }/>);
        const form = wrapper.find(FormRenderer).children().children().instance().form;

        form.change('source.name', 'nameee');
        form.change('source_type', 'openshift');

        setTimeout(() => {
            form.submit().then(() => {
                wrapper.update();
                expect(wrapper.find(FinalWizard)).toHaveLength(1);
                expect(wrapper.find(FinishedStep)).toHaveLength(0);
                expect(wrapper.find(ErroredStep)).toHaveLength(1);
                done();
            });
        }, 1000);
    });
});
