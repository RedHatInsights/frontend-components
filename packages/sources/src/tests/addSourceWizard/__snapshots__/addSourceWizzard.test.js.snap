// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSourceWizard renders correctly with sourceTypes 1`] = `
<SourceAddModal
  applicationTypes={
    Array [
      Object {
        "display_name": "Catalog",
        "id": "1",
        "name": "catalog/overwiev",
      },
      Object {
        "display_name": "Cost Management",
        "id": "2",
        "name": "koku/overwiev",
      },
    ]
  }
  disableAppSelection={false}
  disableHardcodedSchemas={false}
  onCancel={[Function]}
  onSubmit={[Function]}
  sourceTypes={
    Array [
      Object {
        "id": "1",
        "name": "openshift",
        "product_name": "OpenShift Container Platform",
        "schema": Object {
          "fields": Array [
            Object {
              "component": "text-field",
              "name": "field",
            },
          ],
          "title": "openshift",
        },
      },
      Object {
        "id": "2",
        "name": "amazon",
        "product_name": "Amazon Web Services",
        "schema": Object {
          "fields": Array [
            Object {
              "component": "text-field",
              "name": "field",
            },
          ],
          "title": "AWS",
        },
      },
      Object {
        "id": "3",
        "name": "ansible-tower",
        "product_name": "Ansible Tower",
        "schema": Object {
          "fields": Array [
            Object {
              "component": "text-field",
              "name": "field",
            },
          ],
          "title": "AnsibleTower",
        },
      },
    ]
  }
  values={Object {}}
/>
`;

exports[`AddSourceWizard renders correctly without sourceTypes 1`] = `
<SourceAddModal
  applicationTypes={
    Array [
      Object {
        "display_name": "Catalog",
        "id": "1",
        "name": "catalog/overwiev",
      },
      Object {
        "display_name": "Cost Management",
        "id": "2",
        "name": "koku/overwiev",
      },
    ]
  }
  disableAppSelection={false}
  disableHardcodedSchemas={false}
  onCancel={[Function]}
  onSubmit={[Function]}
  values={Object {}}
/>
`;
